extensions-core/lookups-cached-global/src/main/java/org/apache/druid/data/input/MapPopulator.java

```diff
@@ -245,7 +245,8 @@ static long getByteLengthOfObject(@Nullable Object o)
   {
     if (null != o) {
       if (o.getClass().getName().equals(STRING_CLASS_NAME)) {
-        return ((String) (o)).length();
+        // Each String object has ~40 bytes of overhead
+        return ((long) ((String) (o)).length() * Character.BYTES) + 40;
```

Comment:
You could also consider doing this instead:
https://github.com/apache/druid/blob/346fbf133fa338fe0ccf76b5537eb1627f28a487/processing/src/main/java/org/apache/druid/segment/StringDimensionDictionary.java#L42

Reply:
Thanks for pointing this out! The I think depends on the particular version of java and beyond that the particular distribution of it. In my testing I found 40 bytes to be a good estimate for overhead. Sometimes I find that its a bit less. Will leave as is unless you feel strongly about changing.

Reply:
I think we can proceed with what you have. It's a difference of just 4 bytes (40 vs 44) per String anyway, and as you mention, it does depend on java version.

Reply:
it's better to use the same calculation and code. 
@zachjsh - can you reuse the code? The common code can be placed in `StringUtils` class.